To: vim_dev@googlegroups.com
Subject: Patch 7.3.123
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.123
Problem:    ml_get error when executing register being recorded into, deleting
	    lines and 'conceallevel' is set. (ZyX)
Solution:   Don't redraw a line for concealing when it doesn't exist.
Files:	    src/main.c


*** ../vim-7.3.122/src/main.c	2010-12-17 18:06:00.000000000 +0100
--- src/main.c	2011-02-15 16:09:41.000000000 +0100
***************
*** 1198,1204 ****
  			|| conceal_cursor_line(curwin)
  			|| need_cursor_line_redraw))
  	    {
! 		if (conceal_old_cursor_line != conceal_new_cursor_line)
  		    update_single_line(curwin, conceal_old_cursor_line);
  		update_single_line(curwin, conceal_new_cursor_line);
  		curwin->w_valid &= ~VALID_CROW;
--- 1198,1206 ----
  			|| conceal_cursor_line(curwin)
  			|| need_cursor_line_redraw))
  	    {
! 		if (conceal_old_cursor_line != conceal_new_cursor_line
! 			&& conceal_old_cursor_line
! 						<= curbuf->b_ml.ml_line_count)
  		    update_single_line(curwin, conceal_old_cursor_line);
  		update_single_line(curwin, conceal_new_cursor_line);
  		curwin->w_valid &= ~VALID_CROW;
*** ../vim-7.3.122/src/version.c	2011-02-15 15:27:00.000000000 +0100
--- src/version.c	2011-02-15 16:15:28.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     123,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
268. You get up in the morning and go online before getting your coffee.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
